api/clients/v2/relay_payload_retriever.go

```diff
@@ -165,10 +152,10 @@ func (pr *RelayPayloadRetriever) GetPayload(
 			continue
 		}
 
-		payload, err := pr.codec.DecodeBlob(blob)
+		payload, err := blob.ToPayload(pr.config.PayloadPolynomialForm)
 		if err != nil {
 			pr.log.Error(
-				`Cert verification was successful, but decode blob failed!
+				`Cert verification was successful, but conversion from blob to payload failed!
 					This is likely a problem with the local blob codec configuration,
 					but could potentially indicate a maliciously generated blob certificate.
 					It should not be possible for an honestly generated certificate to verify
```

Comment:
can it really be a maliciously generated blob cert? isn't it moreso a bug on the client side?

Reply:
unless you meant maliciously generated blobHeader?